test/hotspot/jtreg/gc/stress/TestStressG1Uncommit.java

```diff
@@ -79,9 +80,9 @@ public static void main(String args[]) throws InterruptedException {
         // Leave 20% head room to try to avoid Full GCs.
         long allocationSize = (long) (Runtime.getRuntime().maxMemory() * 0.8);
 
-        // Figure out suitable number of workers (~1 per gig).
-        int gigsOfAllocation = (int) Math.ceil((double) allocationSize / G);
-        int numWorkers = Math.min(gigsOfAllocation, Runtime.getRuntime().availableProcessors());
+        // Figure out suitable number of workers (~1 per 100M).
+        int allocationChunk = (int) Math.ceil((double) allocationSize * 100 / M);
```

Comment:
If we want to use one worker per 100M, should the equation be `allocationSize / (100 * M)`? Did I miss anything?

Reply:
Yes. It looked weird that he got 10 workers. This test should now always result in `min(8, num_procs)` workers. Especially since this uses `executeLimitedTestJava`, so it will not propagate flags. So I am not sure how we ever run this test with a different heap without editing the test file.

Reply:
Well, "he" made a math mistake that Guixiong noticed. New calculation yields 9 workers on M1, because we take `ceil(0.8*Xmx)`:

```
0s: Using 9 workers, each allocating: ~91M
0s: Allocation size 858993459, allocation chunks: 9
```